File: 2019 icpc ragional practice/Kattis practice/trip.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    double x;
    int n;
    while(cin>>n && n!=0){
        double s[n];
        for(int i = 0; i<n; i++)cin>>s[i];
        double sum = 0.00000000000;
        for(int i=0; i<n; i++)sum += s[i];
        long double ave = sum/ double(n);
        cout<<fixed<<setprecision(5)<<sum<<" "<<ave<<endl;
        sum = 0.0000000000;
        for(int i=0; i<n; i++)sum += abs(s[i] - ave);
        //sum /= 2.0;

        cout<<"$"<<fixed<<setprecision(3)<<sum<<endl;
    }
}

File: 2019 icpc ragional practice/Kattis practice/trip.py
while True:
    n = int(input())
    if n == 0:
        break
    s = []
    for i in range(n):
        s.append(float(input()))
    ave = sum(s) 
    ans = 0
    for i in s:
        ans += abs(n * i - ave)
    print(ans, ave)
    print("$%.2f"%(ans/(2.0*n)))
